openLRSng/TX.h

```diff
@@ -413,12 +664,12 @@ void checkButton(void)
       txWriteEeprom();
     }
 just_bind:
-    // Enter binding mode, automatically after recoding or when pressed for shorter time.
+    // Enter binding mode, automatically after recoding or when pressed for shorter loopTime.
     bindMode();
   }
 }
 
-static inline void checkBND(void)
+void checkBND(void)
```

Comment:
Are these all now used outside of this compilation unit, or why aren't they static any more?

Reply:
From my understanding, there was really no purpose to either the static or inline explicit definitions in this case (or within much of the rest of the code for that matter). 

The inlining should already be taken care of by gcc, both by design and also due to the fact that it's being specified in the makefile.

My understanding is that the static function definition is meant to prevent clashes between compilation units, which isn't really an issue with this code base as it's relatively small & simple. 

Also, Arduino is kind of strange in the sense that we're not actually dealing with source files. Instead, we're dealing with a bunch of header files that all get 'mashed together' anyway (in theory). If we were working with separate .ino or .cpp files that each get compiled separately, then I guess it might make sense to try to help the compiler to optimize & inline things by explicitly defining static functions, But, again, that should already be taken care by gcc of without having to try to influence it.

Anyway, this was my reasoning. It seemed more practical to keep the function definitions clean & consistent and trust in the compiler to handle the low-level details.

Reply:
I totally agree with you that `inline` should not be used explicitly, and instead it should be left to the compiler to optimise this - compilers are quite good at this these days. We should look into making this happen.

Re `static`. You are right that its compile time function of is pretty much rendered useless by the fact that arduino projects tend to stick all the code into header files and mash it together for compilation. (We should probably also look into improving this.) However, there is a second, and just as important function to `static`: It designates a function that has to be treated as private to the compilation unit. i.e. only functions that are part of the API of any given compilation unit should _not_ be `static`. Doing this consequently takes a lot of effort out of doing refactoring, since signature changes to `static` functions only need to change occurrences of this function within the same compilation unit, and can avoid having to do a global search for the function. If compilation units are structured properly (they aren't in arduino), `static` also prevents calls across compilation units in unwanted ways, which is often a cause for bugs.

Reply:
ok, your point is taken, and I agree that if this was a conventional code base, that we'd definitely need to try to keep things organized & separated by compilation unit and API concerns.

However, for the sake of simplicity in this particular project and given the fact that it's still an Arduino/header file mash-up, can we agree to just skip the static/inline directives? (for now at least...)

Reply:
I would like to have the `static` directives reinstated where they were removed. They were added (mostly by me) to improve clarity, so removing them is a regression.

After this discussion, I think we really need to look into refactoring this away from being an arduino project. arduino seems to come with only downsides and no advantages.

Reply:
ok, will reinstate.

I'll try & see if I can move things around into a .cpp/.h setup and see if I can get it to compile without Arduino complaining too much.